File: src/main/java/com/example/smarthire/entities/reclamation/Response.java
package com.example.smarthire.entities.reclamation;

import java.sql.Timestamp;

public class Response {
    private int id;
    private int complaintId;
    private String message;
    private Timestamp responseDate;
    private String adminName;

    public Response() {}

    public Response(int complaintId, String message, String adminName) {
        this.complaintId = complaintId;
        this.message = message;
        this.adminName = adminName;
        this.responseDate = new Timestamp(System.currentTimeMillis());
    }

    // Getters and Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    public int getComplaintId() { return complaintId; }
    public void setComplaintId(int complaintId) { this.complaintId = complaintId; }
    public String getMessage() { return message; }
    public void setMessage(String message) { this.message = message; }
    public Timestamp getResponseDate() { return responseDate; }
    public void setResponseDate(Timestamp responseDate) { this.responseDate = responseDate; }
    public String getAdminName() { return adminName; }
    public void setAdminName(String adminName) { this.adminName = adminName; }
}

File: src/Entity/Response.php
<?php

namespace App\Entity;

use App\Repository\ResponseRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ResponseRepository::class)]
#[ORM\Table(name: 'response')]
class Response
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: Complaint::class, inversedBy: 'responses')]
    #[ORM\JoinColumn(nullable: false, name: 'complaint_id')]
    private ?Complaint $complaint = null;

    #[ORM\Column(type: 'text')]
    private ?string $message = null;

    #[ORM\Column(name: 'response_date', type: 'datetime')]
    private ?\DateTimeInterface $responseDate = null;

    public function __construct()
    {
        $this->responseDate = new \DateTime();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getComplaint(): ?Complaint
    {
        return $this->complaint;
    }

    public function setComplaint(?Complaint $complaint): self
    {
        $this->complaint = $complaint;

        return $this;
    }

    public function getComplaintId(): ?int
    {
        return $this->complaint?->getId();
    }

    public function getMessage(): ?string
    {
        return $this->message;
    }

    public function setMessage(string $message): self
    {
        $this->message = $message;

        return $this;
    }

    public function getResponseDate(): ?\DateTimeInterface
    {
        return $this->responseDate;
    }

    public function setResponseDate(\DateTimeInterface $responseDate): self
    {
        $this->responseDate = $responseDate;

        return $this;
    }
}
